### ISSUE TITLE
Test failures on S390X with 5.7-lkgr

### ISSUE THREAD

**Issue by targos** [nodejs/v8#6]
CI run: https://ci.nodejs.org/job/node-test-commit-linuxone/4064/nodes=rhel72-s390x/

/cc @nodejs/platform-s390

**Comment by bnoordhuis** [comment]
Did you mean 'test failure'?  I don't see a build error in your link but there are some failing tests.

**Comment by targos** [comment]
@bnoordhuis correct. I copy-pasted the title from the other issues.

**Comment by mhdawson** [comment]
@jbajwa has been looking at this today.

**Comment by mhdawson** [comment]
Just gave @jbajwa access, should be able to progress.

**Comment by targos** [comment]
Another try failed while building addon tests: https://ci.nodejs.org/job/node-test-commit-linuxone/4365/nodes=rhel72-s390x/

```
Building addon /data/iojs/build/workspace/node-test-commit-linuxone/nodes/rhel72-s390x/test/addons/null-buffer-neuter/
gyp info it worked if it ends with ok


#
# Fatal error in ../deps/v8/src/runtime/runtime-regexp.cc, line 692
# Check failed: args[1]->IsJSRegExp().
#

==== C stack trace ===============================

    node-gyp(v8::base::debug::StackTrace::StackTrace()+0x1e) [0x8130e516]
    node-gyp(V8_Fatal+0x9c) [0x81308b74]
    node-gyp(v8::internal::Runtime_StringReplaceGlobalRegExpWithString(int, v8::internal::Object**, v8::internal::Isolate*)+0xe4) [0x80e0facc]
    [0x693c104536]
/bin/sh: line 1: 46832 Illegal instruction     (core dumped) env MAKEFLAGS="-j1" ./node deps/npm/node_modules/node-gyp/bin/node-gyp --loglevel=info rebuild --python="/usr/bin/python" --directory="$PWD/$dirname" --nodedir="$PWD"
```

**Comment by jbajwa** [comment]
The above failure seems to be slightly different than what I'm seeing locally and a new run shows the failure I'm seeing (https://ci.nodejs.org/job/node-test-commit-linuxone/nodes=rhel72-s390x/4369/console)

The failures seems to be intermittent, when run locally I am able to have a clean build and some of the testcases fail with:
```
#
# Fatal error in ../deps/v8/src/heap/heap.h, line 457
# Check failed: !object->IsSmi().
#

==== C stack trace ===============================

    node-gyp(v8::base::debug::StackTrace::StackTrace()+0x1e) [0x8130e516]
    node-gyp(V8_Fatal+0x9c) [0x81308b74]
    node-gyp(v8::internal::Heap::AllocateRaw(int, v8::internal::AllocationSpace, v8::internal::AllocationAlignment)+0x266) [0x80b1f636]
    node-gyp(v8::internal::Heap::CopyJSObject(v8::internal::JSObject*, v8::internal::AllocationSite*)+0xb2) [0x80b27162]
    node-gyp(v8::internal::Factory::CopyJSObjectWithAllocationSite(v8::internal::Handle<v8::internal::JSObject>, v8::internal::Handle<v8::internal::AllocationSite>)+0x48) [0x80ae1620]
    node-gyp(v8::internal::JSObjectWalkVisitor<v8::internal::AllocationSiteUsageContext>::StructureWalk(v8::internal::Handle<v8::internal::JSObject>)+0x4cc) [0x80cb0374]
    node-gyp(v8::internal::JSObject::DeepCopy(v8::internal::Handle<v8::internal::JSObject>, v8::internal::AllocationSiteUsageContext*, v8::internal::JSObject::DeepCopyHints)+0x30) [0x80cb0c78]
    node-gyp(v8::internal::Runtime_CreateObjectLiteral(int, v8::internal::Object**, v8::internal::Isolate*)+0x22c) [0x80de7564]
```
All testcases pass when run individually (the failing testcases are part of the parallel bucket).

Although the failure is intermittent but the symptom seems to be consistent. I'm investigating, hoping to have an answer by end of the week.

**Comment by jbajwa** [comment]
I've uploaded the fix to V8 [5.7](https://codereview.chromium.org/2744123002/) (should be landed by tomorrow, waiting for approval). It doesn't apply to V8 5.8 and up because the affected portion of code was moved to TF in [this](https://codereview.chromium.org/2601243003) CL.
Sorry for the delay, took a while to track down the offending instruction.

**Comment by jbajwa** [comment]
The fix is landed on V8 5.7.

**Comment by targos** [comment]
Thank you! I updated the [PR](https://github.com/nodejs/node/pull/11752).
